src/adam_core/observers/state.py

```diff
@@ -8,14 +8,127 @@
 from ..coordinates.cartesian import CartesianCoordinates
 from ..coordinates.origin import Origin, OriginCodes
 from ..time import Timestamp
-from ..utils.spice import get_perturber_state, setup_SPICE
+from ..utils.spice import get_perturber_state, get_spice_body_state, setup_SPICE
 from .observers import OBSERVATORY_CODES, OBSERVATORY_PARALLAX_COEFFICIENTS
 
 R_EARTH_EQUATORIAL = c.R_EARTH_EQUATORIAL
 OMEGA_EARTH = 2 * np.pi / 0.997269675925926
 Z_AXIS = np.array([0, 0, 1])
 
 
+def get_mpc_observer_state(
+    code: str,
+    times: Timestamp,
+    frame: Literal["ecliptic", "equatorial", "itrf93"] = "ecliptic",
+    origin: OriginCodes = OriginCodes.SUN,
+) -> CartesianCoordinates:
+    """
+    Get the state of an MPC observatory on Earth.
+
+    Combines Earth SPICE state and rotation with MPC observatory parallax coefficients.
+
+    Parameters
+    ----------
+    code : str
+        The MPC observatory code of the observer.
+    times : Timestamp
+        The times at which to get the observer state.
+    frame : {'ecliptic', 'equatorial', 'itrf93'}
+        The frame in which to return the observer state.
+    """
+    if frame == "ecliptic":
+        frame_spice = "ECLIPJ2000"
+    elif frame == "equatorial":
+        frame_spice = "J2000"
```

Comment:
Are we missing an IRTF93 check here?

Reply:
Would actually be a funny frame to request an Earth-surface observer state in because it should be nearly constant. I could see it being useful for a satellite or spacecraft though.

Reply:
I guess we could. What would be the "correct" way to do this?

Reply:
I guess you would get the rotational velocity would could be useful, yeah?

Reply:
Yeah that is true, you'd get the rotational velocity. I think as implemented it ought to work.

Reply:
Okay, I've updated with a reviewed AI suggestion. I think it looks correct, though maybe it makes sense to grab earth state higher up to make it less verbose.